packages/orbit-components/src/ListChoice/__tests__/index.test.js

```diff
@@ -1,119 +1,78 @@
-    expect(component.render().prop("data-test")).toBe(dataTest);
+
+  it("should render icon", () => {
+    render(
+      <ListChoice
+        title={title}
+        icon={<Accommodation ariaLabel="Accommodation" />}
+        dataTest={dataTest}
+      />,
+    );
+    const icon = screen.getByLabelText("Accommodation");
+    expect(icon.tagName.toLowerCase()).toBe("svg");
   });
+
   it("should render title and description", () => {
-    expect(content.find("Heading").children().text()).toBe(title);
-    expect(content.find("Text").children().text()).toBe(description);
+    const description = "Further description";
+    render(<ListChoice title={title} description={description} />);
+    expect(screen.getByText(description)).toBeInTheDocument();
   });
+
   it("should render checkbox with checked false", () => {
-    expect(component.find("Checkbox").prop("checked")).toBe(true);
+    render(<ListChoice title={title} selectable />);
+    expect(screen.getByRole("checkbox", { checked: false })).toBeInTheDocument();
   });
+
   it("should NOT execute onClick method", () => {
-    component.simulate("click");
+    render(<ListChoice title={title} disabled onClick={onClick} dataTest={dataTest} />);
+    const component = screen.getByTestId(dataTest);
+    userEvent.click(component);
     expect(onClick).not.toHaveBeenCalled();
   });
+
   it("should execute onClick method", () => {
-    component.setProps({ disabled: false });
-    component.simulate("click");
+    render(<ListChoice title={title} onClick={onClick} dataTest={dataTest} />);
+    const el = screen.getByTestId(dataTest);
+    userEvent.click(el);
     expect(onClick).toHaveBeenCalled();
   });
-  it("should have role button when not selectable", () => {
-    render(
-      <ListChoice
-        title={title}
-        description={description}
-        selectable={false}
-        icon={<Accommodation />}
-        onClick={onClick}
-        dataTest={dataTest}
-      />,
-      screen,
-    );
 
-    const listChoice = screen.querySelector(`[data-test="${dataTest}"]`);
-    expect(listChoice?.getAttribute("role")).toBe("button");
+  it("should have role button when not selectable", () => {
+    render(<ListChoice title={title} onClick={onClick} />);
+    expect(screen.getByRole("button")).toBeInTheDocument();
   });
 
   it("should have role checkbox when selectable", () => {
-    render(
-      <ListChoice
-        title={title}
-        description={description}
-        selectable
-        selected={false}
-        icon={<Accommodation />}
-        onClick={onClick}
-        dataTest={dataTest}
-      />,
-      screen,
-    );
-
-    const listChoice = screen.querySelector(`[data-test="${dataTest}"]`);
-    expect(listChoice?.getAttribute("role")).toBe("checkbox");
-    expect(listChoice?.getAttribute("aria-checked")).toBe("false");
+    render(<ListChoice title={title} selectable dataTest={dataTest} />);
+    expect(screen.getByTestId(dataTest)).toHaveAttribute("role", "checkbox");
```

Comment:
```suggestion
    expect(screen.getByRole("checkbox")).toBeInTheDocument();
```

Reply:
@silvenon  i did it on purpose, for some reason it takes test multiple times, not sure why, if you can help - i'll be happy)

Reply:
This is because both the container and the nested `Checkbox` have the role `checkbox`. I don't think it's necessary for container to imitate a checkbox with `role="checkbox"` and `aria-checked` considering that there is already a `Checkbox` which has these a11y properties naturally, but I'm not sure what's the right thing to do here, maybe add `aria-hidden="true"` to `Checkbox` but it doesn't support it currently, so leave it for later and think about it.